File: 1271.py
class Solution:
    def toHexspeak(self, num: str) -> str:
        res = ""
        mapping = {0:'O', 1:'I', 10:'A', 11:'B', 12:'C', 13:'D', 14:'E', 15:'F'}
        num = int(num)
        while(num != 0):
            digit = num % 16
            num //= 16
            if digit in mapping:
                res += str(mapping[digit])
            else:
                return "ERROR"
        return res[::-1]

File: 1271.cpp
class Solution {
public:
    string toHexspeak(string num) {
        string res = "";
        unordered_map<int, char> mapping;
        mapping[0] = 'O';
        mapping[1] = 'I';
        mapping[10] = 'A';
        mapping[11] = 'B';
        mapping[12] = 'C';
        mapping[13] = 'D';
        mapping[14] = 'E';
        mapping[15] = 'F';
        long long int number = stoll(num);
        while(number != 0){
            if (mapping.find(number % 16) != mapping.end()) {
                res += mapping[number % 16];
            } else {
                return "ERROR";
            }
            number /= 16;
        }
        reverse(res.begin(), res.end());
        return res;
    }
};